relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -71,57 +71,55 @@ fn extract_geo_country_code(event: &Event) -> Option<String> {
+fn is_low_cardinality(source: Option<&TransactionSource>) -> bool {
     match source {
         // For now, we hope that custom transaction names set by users are low-cardinality.
-        TransactionSource::Custom => true,
+        Some(TransactionSource::Custom) => true,
 
         // "url" are raw URLs, potentially containing identifiers.
-        TransactionSource::Url => false,
+        Some(TransactionSource::Url) => false,
 
         // These four are names of software components, which we assume to be low-cardinality.
-        TransactionSource::Route
-        | TransactionSource::View
-        | TransactionSource::Component
-        | TransactionSource::Task => true,
+        Some(TransactionSource::Route)
+        | Some(TransactionSource::View)
+        | Some(TransactionSource::Component)
+        | Some(TransactionSource::Task) => true,
 
         // We know now that the rules to remove high cardinality were applied, so we assume
         // low-cardinality now.
-        TransactionSource::Sanitized => true,
+        Some(TransactionSource::Sanitized) => true,
 
-        // "unknown" is the value for old SDKs that do not send a transaction source yet.
-        // Caller decides how to treat this.
-        TransactionSource::Unknown => treat_unknown_as_low_cardinality,
+        // Explicit `Unknown` is used to mark a legacy SDK that does not send the transaction name,
+        // but we assume sends low-cardinality data. See `is_high_cardinality_transaction`.
+        Some(TransactionSource::Unknown) => true,
 
         // Any other value would be an SDK bug or users manually configuring the
         // source, assume high-cardinality and drop.
-        TransactionSource::Other(_) => false,
+        Some(TransactionSource::Other(_)) => false,
+
+        // `None` is used to mark a legacy SDK that does not send the transaction name,
+        // and we assume sends high-cardinality data. See `is_high_cardinality_transaction`.
+        None => false,
     }
 }
 
 /// Decide whether we want to keep the transaction name.
 /// High-cardinality sources are excluded to protect our metrics infrastructure.
 /// Note that this will produce a discrepancy between metrics and raw transaction data.
-fn get_transaction_name(
-    event: &Event,
-    accept_transaction_names: AcceptTransactionNames,
-) -> Option<String> {
+fn get_transaction_name(event: &Event) -> Option<String> {
     let original_transaction_name = match event.transaction.value() {
         Some(name) => name,
         None => {
             return None;
         }
     };
 
-    // In client-based mode, handling of "unknown" sources depends on the SDK name.
-    // In strict mode, treat "unknown" as high cardinality.
-    let treat_unknown_as_low_cardinality = matches!(
-        accept_transaction_names,
-        AcceptTransactionNames::ClientBased
-    ) && !store::is_high_cardinality_sdk(event);
```

Comment:
No need to call `is_high_cardinality_sdk` again (already done in normalization). Just rely on the value of `event.transaction_info.source`.